src/wallet/components/WalletDropdownFundLink.tsx

```diff
@@ -0,0 +1,74 @@
+import { isValidElement, useMemo } from 'react';
+import { FundWalletSvg } from '../../internal/svg/fundWallet';
+import { cn, pressable, text as themeText } from '../../styles/theme';
+import type { WalletDropdownFundLinkReact } from '../types';
+
+export function WalletDropdownFundLink({
+  className,
+  icon,
+  rel,
+  target,
+  text = 'Deposit Funds',
+  type = 'tab',
+  size = 'm',
+}: WalletDropdownFundLinkReact) {
+  const iconSvg = useMemo(() => {
+    if (icon === undefined) {
+      return FundWalletSvg;
+    }
+    if (isValidElement(icon)) {
+      return icon;
+    }
+  }, [icon]);
+
+  const currentURL = window.location.href;
+  const tabName = document.title;
+  const windowSizes: Record<
+    's' | 'm' | 'l',
+    {
+      width: number;
+      height: number;
+    }
+  > = {
+    s: { width: 400, height: 500 },
+    m: { width: 600, height: 700 },
+    l: { width: 800, height: 900 },
+  };
+
+  const handleClick = (e: React.MouseEvent) => {
```

Comment:
All this logic should be his own utility.

Reply:
most [handleClick](https://github.com/search?q=repo%3Acoinbase%2Fonchainkit%20handleclick&type=code) in our repo are in the same component, do we really need a separate `util` file for just formatting the link? 

it's 6 lines of code - happy to do so if this is a new standard moving forward

Reply:
What needs to be in his own util are isolate `input` / `output` operations.

In this case the `getWindowDetails` is his own utility.

Plus will make the test and code, easier to write and mantain.